### PR TITLE
(CDAP-2988) Added tests for Spark using getDataset() directly.

### PR DESCRIPTION
**Pull request by bdmogal**
Also removed some unused code.

Build: http://builds.cask.co/browse/CDAP-DUT2910-1

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java [hunk 1]
```diff
@@ -396,6 +400,122 @@ private void testSparkWithCustomFileSet(Class<?> appClass, Class<?> programClass
...
+
+    KeyValueTable logStatsTable = datasetInstantiator.getDataset("logStats");
+    validateGetDatasetOutput(logStatsTable);
+  }
+
+  private void validateGetDatasetOutput(KeyValueTable logStatsTable) {
+    CloseableIterator<KeyValue<byte[], byte[]>> scan = logStatsTable.scan(null, null);
+    try {
+      // must have 4 records
+      Assert.assertTrue(scan.hasNext());
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java]: Can we do something like this: 

```
      int counter = 0;
      while (counter < 4) {
        Assert.assertTrue("Does not have" + counter + "value", scan.hasNext());
        validateKeyValue(scan.next());
        counter++;
      }
```

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java [hunk 2]
```diff
@@ -396,6 +400,122 @@ private void testSparkWithCustomFileSet(Class<?> appClass, Class<?> programClass
...
+    SparkAppUsingGetDataset.LogStats fredStats2 = new SparkAppUsingGetDataset.LogStats(1, 50);
+    SparkAppUsingGetDataset.LogStats bradStats1 = new SparkAppUsingGetDataset.LogStats(1, 50);
+    SparkAppUsingGetDataset.LogStats bradStats2 = new SparkAppUsingGetDataset.LogStats(1, 50);
+
+    SparkAppUsingGetDataset.LogKey logKey =
+      GSON.fromJson(Bytes.toString(keyValue.getKey()), SparkAppUsingGetDataset.LogKey.class);
+    SparkAppUsingGetDataset.LogStats logStats =
+      GSON.fromJson(Bytes.toString(keyValue.getValue()), SparkAppUsingGetDataset.LogStats.class);
+
+    if (fredKey1.equals(logKey)) {
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java]: I think this if else loop can be simplified a bit if you put the above in a Map and then do contains key with logKey in then get if found to compare value else your assert fail.

**Comment by rohitsinha54** [comment]
Two comments besides that LGTM for merge after build passes.

**Comment by rohitsinha54** [comment]
LGTM

**Comment by bdmogal** [comment]
Build green. Squashed commits. Merging.

**Pull request merged**